File: src/app/youtube.service.ts
import { Injectable } from '@angular/core';
import { GetPlayListResponse } from './models/youtube.models';

@Injectable({
  providedIn: 'root'
})
export class YoutubeService {

  baseUrl = 'https://www.googleapis.com/youtube/v3/'
  channelId = 'UC5mO6RcazceHz-FoWYKyDRw'
  key = 'AIzaSyAeOSgy3k8btZG-k-D3zdgU537UY3TkHa4'

  constructor() { }

  getSermons(pageSize: number, nextPageToken: string) : Promise<GetPlayListResponse> {
    const requestOptions = {
        method: 'GET',
        headers: { 'Content-Type': 'application/json' }
    };

    let url = `${this.baseUrl}search?part=snippet&maxResults=${pageSize}&type=video&order=date&channelId=${this.channelId}&key=${this.key}`
    if (nextPageToken) {
      url += `&pageToken=${nextPageToken}`
    }
    
    return fetch(url, requestOptions)
        .then(this.handleResponse)
        .then(response => {
            return response;
        })
        .catch(error => {
        });
  }

  getSeries(pageSize: number, nextPageToken: string) : Promise<GetPlayListResponse> {
    const requestOptions = {
        method: 'GET',
        headers: { 'Content-Type': 'application/json' }
    };

    let url = `${this.baseUrl}playlists?part=snippet&maxResults=${pageSize}&type=video&order=date&channelId=${this.channelId}&key=${this.key}`
    if (nextPageToken) {
      url += `&pageToken=${nextPageToken}`
    }
    
    return fetch(url, requestOptions)
        .then(this.handleResponse)
        .then(response => {
            return response;
        })
        .catch(error => {
        });
  }

  getLiveEvent() : Promise<GetPlayListResponse> {
    const requestOptions = {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' }
    };

    let url = `${this.baseUrl}search?part=snippet&eventType=live&type=video&channelId=${this.channelId}&key=${this.key}`

    return fetch(url, requestOptions)
        .then(this.handleResponse)
        .then(response => {
            return response;
        })
        .catch(error => {
        });
  }

  handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
  }
}

File: aws_api/SermonsAPI/Services/Youtube/Implementations/YouTubeService.cs
using Config.Interfaces;
using GlobalArticleDatabaseAPI.Models;
using GlobalArticleDatabaseAPI.Services.Youtube.Interface;
using Newtonsoft.Json;
using System;
using System.Net.Http;
using System.Threading.Tasks;

namespace GlobalArticleDatabaseAPI.Services.Youtube.Implementations
{
  public class YouTubeService : IYouTubeService
  {
    private readonly ISettings _settings;
    private readonly HttpClient _client;
    public YouTubeService(ISettings settings)
    {
      _settings = settings ?? throw new ArgumentNullException(nameof(settings));

      _client = new HttpClient();
      _client.BaseAddress = new Uri("https://www.googleapis.com/youtube/v3/");
    }

    public async Task<GetItemsResponse> GetLiveEvent()
    {
      var url = $"search?part=snippet&type=video&eventType=live&channelId={_settings.YouTubeChannelId}&key={_settings.YouTubeKey}";

      var response = await _client.GetAsync(url);

      return JsonConvert.DeserializeObject<GetItemsResponse>(await response.Content.ReadAsStringAsync());
    }

    public async Task<GetItemsResponse> GetSeries(string pageToken)
    {
      var url = $"playlists?part=snippet&order=date&maxResults=50&channelId={_settings.YouTubeChannelId}&key={_settings.YouTubeKey}";
      if (!string.IsNullOrEmpty(pageToken))
      {
        url += $"pageToken={pageToken}";
      }

      var response = await _client.GetAsync(url);

      return JsonConvert.DeserializeObject<GetItemsResponse>(await response.Content.ReadAsStringAsync());
    }

    public async Task<GetItemsResponse> GetSermons(string pageToken)
    {
      var url = $"playlistItems?part=snippet&playlistId={_settings.SermonsPlayListId}&maxResults=50&channelId={_settings.YouTubeChannelId}&key={_settings.YouTubeKey}";
      if (!string.IsNullOrEmpty(pageToken ))
      {
        url += $"&pageToken={pageToken}";
      }

      var response = await _client.GetAsync(url);

      var data = JsonConvert.DeserializeObject<GetItemsResponse>(await response.Content.ReadAsStringAsync());

      data?.items?.ForEach(item =>
      {
        if (item!= null && item.snippet!= null && item.snippet.description != null)
        {
          var lines = item.snippet.description.Split(new string[] { "\n" }, StringSplitOptions.None);
          if (lines.Length > 1)
          {            
            if (DateTime.TryParse(lines[1], out var date))
            {
              item.snippet.description = lines[0];
              item.snippet.publishedAt = date;
            }
          }
        }        
      });

      return data;
    }
  }
}
